yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/call/docs/DocScanService.java

```diff
@@ -0,0 +1,277 @@
+package com.yoti.api.client.spi.remote.call.docs;
+
+import static com.yoti.api.client.spi.remote.call.HttpMethod.HTTP_DELETE;
+import static com.yoti.api.client.spi.remote.call.HttpMethod.HTTP_GET;
+import static com.yoti.api.client.spi.remote.call.HttpMethod.HTTP_POST;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.*;
+import static com.yoti.api.client.spi.remote.util.Validation.notNull;
+import static com.yoti.api.client.spi.remote.util.Validation.notNullOrEmpty;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.util.List;
+
+import com.yoti.api.client.Media;
+import com.yoti.api.client.docs.DocScanException;
+import com.yoti.api.client.docs.session.create.CreateSessionResult;
+import com.yoti.api.client.docs.session.create.SessionSpec;
+import com.yoti.api.client.docs.session.create.SimpleCreateSessionResult;
+import com.yoti.api.client.docs.session.retrieve.DocScanSession;
+import com.yoti.api.client.spi.remote.MediaValue;
+import com.yoti.api.client.spi.remote.call.*;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.yoti.api.client.spi.remote.call.factory.UnsignedPathFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service built to handle the interactions between the client and
+ * Doc Scan API's.
+ *
+ * @since 2.7
+ */
+public final class DocScanService {
+
+    public static DocScanService newInstance() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+        return new DocScanService(new UnsignedPathFactory(), objectMapper);
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(DocScanService.class);
+
+    private final UnsignedPathFactory unsignedPathFactory;
+    private final ObjectMapper objectMapper;
+
+    private final String apiUrl;
+
+    private DocScanService(UnsignedPathFactory pathFactory, ObjectMapper objectMapper) {
+        this.unsignedPathFactory = pathFactory;
+        this.objectMapper = objectMapper;
+
+        apiUrl = System.getProperty(PROPERTY_YOTI_DOCS_URL, DEFAULT_YOTI_DOCS_URL);
```

Comment:
Perhaps we should consider passing this into the service as configuration/parameter instead of setting globally, as it's something integrators will need to change when using the sandbox.

It also looks like this uses the same `PROPERTY_YOTI_DOCS_URL` key as the sandbox (same applies for `PROPERTY_YOTI_DOCS_URL`) - could this cause conflicts or are they encapsulated in some way?

Also related to this, perhaps we should consider a separate client for doc scan as it has a different base URL and could potentially cause complications in other SDKs that provide the base URL to the clients using constructor or setter (we'd need to pass in a base URL for each service). As an example the [AWS SDK](https://github.com/aws/aws-sdk-java) takes this approach.

Thoughts on this @echarrod ?

Reply:
I'm a fan of having separate clients to avoid any conflicts, and also make it cleaner. This will be a problem in some SDKs more than others, but it would be ideal to adopt a common approach across them all.

Reply:
Separate clients have been created